src/components/vanilla/Button.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Spinner from './Spinner';
+import { REGULAR_FONT_SIZE } from '../constants';
+
+export default function Button({ buttonLabel, showSpinner, disabled, onClick, icon }: { buttonLabel?: string; showSpinner?: boolean; disabled?: boolean;}) {
```

Comment:
lets create a type for this please

`type ButtonProps = { buttonLabel?: string; showSpinner?: boolean; disabled?: boolean;}`

Reply:
@mad-raccoon do you mean instead of writing it how it is now? Or do you mean add a type for onClick and icon?

Button({ buttonLabel, showSpinner, disabled, onClick, icon }: { buttonLabel?: string; showSpinner?: boolean; disabled?: boolean;})

Reply:
In any case, updating to:

type OnClick = (event: React.MouseEvent<HTMLButtonElement>) => void;

interface ButtonProps {
  buttonLabel?: string;
  showSpinner?: boolean;
  disabled?: boolean;
  onClick?: OnClick;
  icon?: React.ReactNode;
}


export default function Button({ buttonLabel, showSpinner, disabled, onClick, icon }: ButtonProps) {